unstructured_ingest/v2/processes/connectors/github.py

```diff
@@ -0,0 +1,238 @@
+                }
+                if self.connection_config.branch is not None:
+                    record_locator["branch"] = self.connection_config.branch
+
+                date_modified = datetime.strptime(
+                    element._headers["last-modified"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                date_created = datetime.strptime(
+                    element._headers["date"],
+                    "%a, %d %b %Y %H:%M:%S %Z",
+                ).isoformat()
+
+                yield FileData(
+                    identifier=element.sha,
+                    connector_type=CONNECTOR_TYPE,
+                    source_identifiers=SourceIdentifiers(
+                        fullpath=element.path,
+                        filename=element.path.split("/")[-1],
+                        rel_path=rel_path,
+                    ),
+                    metadata=FileDataSourceMetadata(
+                        url=element.url,
+                        version=element.etag,
+                        record_locator=record_locator,
+                        date_modified=date_modified,
+                        date_created=date_created,
+                        date_processed=str(time()),
+                    ),
+                    additional_metadata={
+                        "content-type": element._headers.get("content-type", ""),
+                        "mode": element._rawData.get("mode", ""),
+                        "type": element._rawData.get("type", ""),
+                        "size": element._rawData.get("size", ""),
+                    },
+                )
+
+
+class GitHubDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class GitHubDownloader(Downloader):
+    connection_config: GitHubConnectionConfig
+    download_config: GitHubDownloaderConfig
+
+    @requires_dependencies(["github"], extras="github")
+    def _fetch_file(self, path):
+        from github.GithubException import UnknownObjectException
+
+        try:
+            content_file = self.connection_config.get_repo().get_contents(path)
+        except UnknownObjectException:
+            logger.error(f"File doesn't exists {self.connection_config.url}/{path}")
+            raise UnknownObjectException
+
+        return content_file
+
+    @SourceConnectionNetworkError.wrap
+    @requires_dependencies(["requests"], extras="github")
+    def _fetch_content(self, content_file):
+        import requests
+        from github.GithubException import UnknownObjectException
+
+        contents = b""
+        if (
+            not content_file.content  # type: ignore
+            and content_file.encoding == "none"  # type: ignore
+            and content_file.size  # type: ignore
+        ):
+            logger.info("File too large for the GitHub API, using direct download link instead.")
+            response = requests.get(content_file.download_url)  # type: ignore
```

Comment:
Given that this is just using the request library, we could swap this for [httpx ](https://www.python-httpx.org/) to get async support.